### ISSUE TITLE
[help wanted] any way to update learning rate and momentum

### ISSUE THREAD

**Issue by fyears** [fchollet/keras#109]
Hi, I am reading [a tutorial with nolearn/lasagne](http://danielnouri.org/notes/2014/12/17/using-convolutional-neural-nets-to-detect-facial-keypoints-tutorial/), where the author point out it is possible to [update learning rate and momentum over time](http://danielnouri.org/notes/2014/12/17/using-convolutional-neural-nets-to-detect-facial-keypoints-tutorial/#changing-learning-rate-and-momentum-over-time).

Is it possible to do it in keras? The coverage speed is low, so I am wondering whether it's possible to use the technique to improve the speed.

**Comment by fchollet** [comment]
Look at the `decay` parameter in SGD to get a decreasing learning rate over time.

Otherwise, you can manually set `sgd.momentum`, `sgd.lr` and `sgd.decay` (where `sgd` is an instance of `SGD`). Note that `sgd.lr` is the *base* learning rate, not the actual learning rate, ie. it is the learning rate before decay is applied.

**Comment by chasep255** [comment]
How is the decay applied to the learning rate?  Is it multiplied by the decay factor after each epoch?  Is the decay subtracted from it?  Is it exponential?
